test/protocol-tests-core/src/main/java/software/amazon/awssdk/protocol/reflect/ShapeModelReflector.java

```diff
@@ -221,6 +221,8 @@ private Object getSimpleMemberValue(JsonNode currentNode, MemberModel memberMode
         switch (memberModel.getVariable().getSimpleType()) {
             case "Long":
                 return currentNode.asLong();
+            case "Short":
```

Comment:
Why asDouble and not .asInt(); ?

Reply:
No preference, can change if needed.

Reply:
I was also thinking of isShort check  however since its just in tests I think no need to jsonNode.isShort check. For now since int is closest to short  in the possible asMethod options we can use asInt().

Reply:
Sure fixed.